test/unit/controllers/DeclarationsDocumentationControllerSpec.scala

```diff
@@ -0,0 +1,163 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package unit.controllers
+
+import org.mockito.Mockito.reset
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.mockito.MockitoSugar
+import org.scalatestplus.play._
+import play.api.http.HttpErrorHandler
+import play.api.libs.json.Json
+import play.api.mvc._
+import play.api.test._
+import play.api.test.Helpers._
+import uk.gov.hmrc.customs.declaration.controllers.DeclarationsDocumentationController
+
+class DeclarationsDocumentationControllerSpec extends PlaySpec with MockitoSugar with Results with BeforeAndAfterEach {
+
+  private val mockService = mock[HttpErrorHandler]
+
+  private val v1WhitelistedAppIdsConfigs = Map(
+    "api.access.version-1.0.whitelistedApplicationIds.0" -> "v1AppId-1",
+    "api.access.version-1.0.whitelistedApplicationIds.1" -> "v1AppId-2")
+
+  private val v2WhitelistedAppIdsConfigs = Map(
+    "api.access.version-2.0.whitelistedApplicationIds.0" -> "v2AppId-1",
+    "api.access.version-2.0.whitelistedApplicationIds.1" -> "v2AppId-2")
+
+  private def getApiDefinitionWith(configMap: Map[String, Any]) =
+    new DeclarationsDocumentationController(mockService, play.api.Configuration.from(configMap))
+      .definition()
+
+  override def beforeEach() {
+    reset(mockService)
+  }
+
+  "API Definition" should {
+
+    "is correct when V1 & V2 are PUBLIC by default" in {
+      val result = getApiDefinitionWith(Map())(FakeRequest())
+
+      status(result) mustBe 200
+      contentAsJson(result) mustBe expectedJson(None, None)
+    }
+
+    "is correct when V1 is PRIVATE & V2 is public" in {
+      val result = getApiDefinitionWith(v1WhitelistedAppIdsConfigs)(FakeRequest())
+
+      status(result) mustBe 200
+      contentAsJson(result) mustBe expectedJson(expectedV1WhitelistedAppIds = Some(v1WhitelistedAppIdsConfigs.values), None)
+    }
+
+    "is correct when V1 is PUBLIC & V2 is PRIVATE" in {
+      val result = getApiDefinitionWith(v2WhitelistedAppIdsConfigs)(FakeRequest())
+
+      status(result) mustBe 200
+      contentAsJson(result) mustBe expectedJson(expectedV1WhitelistedAppIds = None, expectedV2WhitelistedAppIds = Some(v2WhitelistedAppIdsConfigs.values))
+    }
+
+    "is correct when V1 & V2 are PRIVATE" in {
+      val result = getApiDefinitionWith(v1WhitelistedAppIdsConfigs ++ v2WhitelistedAppIdsConfigs)(FakeRequest())
+
+      status(result) mustBe 200
+      contentAsJson(result) mustBe expectedJson(
```

Comment:
space needed `=Some(v1WhitelistedAppIdsConfigs.values)` should be `= Some(v1WhitelistedAppIdsConfigs.values)`

Reply:
👍